src/tasks/gitea/gitea-drone-oauth.ts

```diff
@@ -0,0 +1,212 @@
+      .map((c) => c.trim())
+      .filter((cookie) => {
+        return cookie.includes(this.csrfCookieName) // Find the key/value pair with _csrf as key
+      })
+      .map((cookie) => {
+        return cookie.substring(this.csrfCookieName.length + 1) // Retrieve value for '_csrf'-key
+      })[0]
+
+    console.log('Granting authorization')
+
+    const grantOptions: AxiosRequestConfig = {
+      method: 'POST',
+      url: `${this.giteaUrl}/login/oauth/grant`,
+      headers: {
+        cookie: authorizeHeaderCookies
+          .map((cookie) => {
+            return cookie.split(';')[0]
+          })
+          .join('; '),
+        Cookie: authorizeHeaderCookies
+          .map((cookie) => {
+            return cookie.split(';')[0]
+          })
+          .join('; '),
+      },
+      maxRedirects: 1,
+      auth: {
+        username: env.GITEA_USER,
+        password: env.GITEA_PASSWORD,
+      },
+      // Data for this post query must be stringified https://github.com/axios/axios#using-applicationx-www-form-urlencoded-format
+      data: querystring.stringify({
+        _csrf: this.csrfToken,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        client_id: `${this.oauthData.clientId}`,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        redirect_uri: `${this.droneUrl}/login`,
+      }),
+    }
+
+    try {
+      await axios.request(grantOptions)
+    } catch (error) {
+      // Do nothing, error code could be on the redirect
+    }
+  }
+
+  async main() {
+    const client = this.getApiClient()
+
+    const user = new UserApi(env.GITEA_USER, env.GITEA_PASSWORD, `${this.giteaUrl}/api/v1`)
+    const oauthOpts = new CreateOAuth2ApplicationOptions()
+    oauthOpts.name = 'otomi-drone'
+    oauthOpts.redirectUris = [`${this.droneUrl}/login`]
+    try {
+      const remoteSecret = await this.secretExists()
+      const oauthApps: OAuth2Application[] = (await user.userGetOauth2Application()).body.filter(
+        (x) => x.name === oauthOpts.name,
+      )
+      // If secret exists (correctly) and oauth apps are (still) defined, everything is good
+      if (
+        remoteSecret &&
+        remoteSecret.clientId.length > 0 &&
+        remoteSecret.clientSecret.length > 0 &&
+        oauthApps.length > 0 &&
+        oauthApps[0].clientId === Buffer.from(remoteSecret.clientId, 'base64').toString()
+      ) {
+        console.log('Gitea Drone OAuth secret exists')
+        this.oauthData = {
+          clientId: Buffer.from(remoteSecret.clientId, 'base64').toString(),
+          clientSecret: Buffer.from(remoteSecret.clientSecret, 'base64').toString(),
+        }
+        await this.authorizeOAuth()
+        return
+      }
+      console.log('Some data did not match, recreating OAuth data')
+
+      // Otherwise, clear (if necessary)
+      try {
+        await client.deleteNamespacedSecret(this.k8s_secret_name, this.k8s_namespace)
+      } catch (e) {
+        // Secret didn't exist in
+      }
```

Comment:
here we can allow try catch because it avoids an extra call to check for the secrets existence